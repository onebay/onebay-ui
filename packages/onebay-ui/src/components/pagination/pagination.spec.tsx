import { mount } from '@vue/test-utils'
import Pagination from './index'

const factory = (values = {}, slots = { default: [] }) => {
  return mount(Pagination, {
    components: {},
    slots,
    props: { ...values },
  })
}

describe('Pagination Snap', () => {
  it('render Pagination -- props current', () => {
    const wrapper = factory({ current: 2 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Pagination -- props total', () => {
    const wrapper = factory({ total: 100 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Pagination -- props pageSize', () => {
    const wrapper = factory({ pageSize: 40 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Pagination -- props icon', () => {
    const wrapper = factory({ icon: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Pagination -- props total change', async () => {
    const wrapper = factory({ icon: true })
    await wrapper.setProps({ total: 30 })
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('Pagination Event', () => {
  it('Pagination onPageChange - prev', () => {
    const onPageChange = jest.fn()
    const wrapper = factory({
      total: 100,
      current: 2,
      onPageChange,
    })
    expect(wrapper.element).toMatchSnapshot()
    wrapper.find('.ob-pagination .ob-pagination__btn-prev .ob-button').trigger('click')
    expect(onPageChange).toBeCalled()
    expect(wrapper.vm.state.currentPage).toBe(1)
    wrapper.find('.ob-pagination .ob-pagination__btn-prev .ob-button').trigger('click')
    expect(wrapper.vm.state.currentPage).toBe(1)
  })

  it('Pagination onPageChange - next', () => {
    const onPageChange = jest.fn()
    const wrapper = factory({
      total: 60,
      current: 2,
      onPageChange,
    })
    expect(wrapper.element).toMatchSnapshot()
    wrapper.find('.ob-pagination .ob-pagination__btn-next .ob-button').trigger('click')
    expect(onPageChange).toBeCalled()
    expect(wrapper.vm.state.currentPage).toBe(3)
    wrapper.find('.ob-pagination .ob-pagination__btn-next .ob-button').trigger('click')
    expect(wrapper.vm.state.currentPage).toBe(3)
  })

  it('Pagination onPageChange not to be called(disabled prev or next)', () => {
    const onPageChange = jest.fn()
    const wrapper = factory({
      total: 20,
      onPageChange,
    })
    expect(wrapper.element).toMatchSnapshot()
    wrapper.find('.ob-pagination .ob-pagination__btn-prev .ob-button').trigger('click')
    wrapper.find('.ob-pagination .ob-pagination__btn-next .ob-button').trigger('click')
    expect(onPageChange).not.toBeCalled()
  })

  it('Pagination onPageChange params {type, current} - prev', () => {
    const onPageChange = jest.fn()
    const wrapper = factory({
      current: 2,
      total: 100,
      onPageChange,
    })
    wrapper.find('.ob-pagination .ob-pagination__btn-prev .ob-button').trigger('click')
    expect(onPageChange.mock.calls[0][0].type).toEqual('prev')
    expect(onPageChange.mock.calls[0][0].current).toEqual(1)
  })

  it('Pagination onPageChange params {type, current} - next', () => {
    const onPageChange = jest.fn()
    const wrapper = factory({
      total: 100,
      onPageChange,
    })
    wrapper.find('.ob-pagination .ob-pagination__btn-next .ob-button').trigger('click')
    expect(onPageChange.mock.calls[0][0].type).toEqual('next')
    expect(onPageChange.mock.calls[0][0].current).toEqual(2)
  })
})
