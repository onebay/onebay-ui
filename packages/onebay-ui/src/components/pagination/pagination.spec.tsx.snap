// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Event Pagination onPageChange - next 1`] = `
<div
  class="ob-pagination"
>
  <div
    class="ob-pagination__btn-prev"
  >
    <div
      class="ob-button ob-button--small"
    >
      <!---->
      <div
        class="ob-button__text"
      >
        
        pre
        
      </div>
    </div>
  </div>
  <div
    class="ob-pagination__number"
  >
    <span
      class="ob-pagination__number-current text"
    >
      2
    </span>
    /
    3
  </div>
  <div
    class="ob-pagination__btn-next"
  >
    <div
      class="ob-button ob-button--small"
    >
      <!---->
      <div
        class="ob-button__text"
      >
        
        next
        
      </div>
    </div>
  </div>
</div>
`;

exports[`Pagination Event Pagination onPageChange - prev 1`] = `
<div
  class="ob-pagination"
>
  <div
    class="ob-pagination__btn-prev"
  >
    <div
      class="ob-button ob-button--small"
    >
      <!---->
      <div
        class="ob-button__text"
      >
        
        pre
        
      </div>
    </div>
  </div>
  <div
    class="ob-pagination__number"
  >
    <span
      class="ob-pagination__number-current text"
    >
      2
    </span>
    /
    5
  </div>
  <div
    class="ob-pagination__btn-next"
  >
    <div
      class="ob-button ob-button--small"
    >
      <!---->
      <div
        class="ob-button__text"
      >
        
        next
        
      </div>
    </div>
  </div>
</div>
`;

exports[`Pagination Event Pagination onPageChange not to be called(disabled prev or next) 1`] = `
<div
  class="ob-pagination"
>
  <div
    class="ob-pagination__btn-prev"
  >
    <div
      class="ob-button ob-button--small ob-button--disabled"
    >
      <!---->
      <div
        class="ob-button__text"
      >
        
        pre
        
      </div>
    </div>
  </div>
  <div
    class="ob-pagination__number"
  >
    <span
      class="ob-pagination__number-current text"
    >
      1
    </span>
    /
    1
  </div>
  <div
    class="ob-pagination__btn-next"
  >
    <div
      class="ob-button ob-button--small ob-button--disabled"
    >
      <!---->
      <div
        class="ob-button__text"
      >
        
        next
        
      </div>
    </div>
  </div>
</div>
`;

exports[`Pagination Snap render Pagination -- props current 1`] = `
<div
  class="ob-pagination"
>
  <div
    class="ob-pagination__btn-prev"
  >
    <div
      class="ob-button ob-button--small ob-button--disabled"
    >
      <!---->
      <div
        class="ob-button__text"
      >
        
        pre
        
      </div>
    </div>
  </div>
  <div
    class="ob-pagination__number"
  >
    <span
      class="ob-pagination__number-current text"
    >
      2
    </span>
    /
    1
  </div>
  <div
    class="ob-pagination__btn-next"
  >
    <div
      class="ob-button ob-button--small ob-button--disabled"
    >
      <!---->
      <div
        class="ob-button__text"
      >
        
        next
        
      </div>
    </div>
  </div>
</div>
`;

exports[`Pagination Snap render Pagination -- props icon 1`] = `
<div
  class="ob-pagination ob-pagination--icon"
>
  <div
    class="ob-pagination__btn-prev"
  >
    <div
      class="ob-button ob-button--small ob-button--disabled"
    >
      <!---->
      <div
        class="ob-button__text"
      >
        
        <div
          class="ob-icon ob-icon-chevron-left text"
        />
        
      </div>
    </div>
  </div>
  <div
    class="ob-pagination__number"
  >
    <span
      class="ob-pagination__number-current text"
    >
      1
    </span>
    /
    1
  </div>
  <div
    class="ob-pagination__btn-next"
  >
    <div
      class="ob-button ob-button--small ob-button--disabled"
    >
      <!---->
      <div
        class="ob-button__text"
      >
        
        <div
          class="ob-icon ob-icon-chevron-right text"
        />
        
      </div>
    </div>
  </div>
</div>
`;

exports[`Pagination Snap render Pagination -- props pageSize 1`] = `
<div
  class="ob-pagination"
>
  <div
    class="ob-pagination__btn-prev"
  >
    <div
      class="ob-button ob-button--small ob-button--disabled"
    >
      <!---->
      <div
        class="ob-button__text"
      >
        
        pre
        
      </div>
    </div>
  </div>
  <div
    class="ob-pagination__number"
  >
    <span
      class="ob-pagination__number-current text"
    >
      1
    </span>
    /
    1
  </div>
  <div
    class="ob-pagination__btn-next"
  >
    <div
      class="ob-button ob-button--small ob-button--disabled"
    >
      <!---->
      <div
        class="ob-button__text"
      >
        
        next
        
      </div>
    </div>
  </div>
</div>
`;

exports[`Pagination Snap render Pagination -- props total 1`] = `
<div
  class="ob-pagination"
>
  <div
    class="ob-pagination__btn-prev"
  >
    <div
      class="ob-button ob-button--small ob-button--disabled"
    >
      <!---->
      <div
        class="ob-button__text"
      >
        
        pre
        
      </div>
    </div>
  </div>
  <div
    class="ob-pagination__number"
  >
    <span
      class="ob-pagination__number-current text"
    >
      1
    </span>
    /
    5
  </div>
  <div
    class="ob-pagination__btn-next"
  >
    <div
      class="ob-button ob-button--small"
    >
      <!---->
      <div
        class="ob-button__text"
      >
        
        next
        
      </div>
    </div>
  </div>
</div>
`;

exports[`Pagination Snap render Pagination -- props total change 1`] = `
<div
  class="ob-pagination ob-pagination--icon"
>
  <div
    class="ob-pagination__btn-prev"
  >
    <div
      class="ob-button ob-button--small ob-button--disabled"
    >
      <!---->
      <div
        class="ob-button__text"
      >
        
        <div
          class="ob-icon ob-icon-chevron-left text"
        />
        
      </div>
    </div>
  </div>
  <div
    class="ob-pagination__number"
  >
    <span
      class="ob-pagination__number-current text"
    >
      1
    </span>
    /
    2
  </div>
  <div
    class="ob-pagination__btn-next"
  >
    <div
      class="ob-button ob-button--small"
    >
      <!---->
      <div
        class="ob-button__text"
      >
        
        <div
          class="ob-icon ob-icon-chevron-right text"
        />
        
      </div>
    </div>
  </div>
</div>
`;
